src/20240229_AaveV3Ethereum_AmendSafetyModuleEmissions/AaveV3Ethereum_AmendSafetyModuleEmissions_20240229.sol

```diff
@@ -0,0 +1,67 @@
+
+import {AaveV3Ethereum, AaveV3EthereumAssets} from 'aave-address-book/AaveV3Ethereum.sol';
+import {MiscEthereum} from 'aave-address-book/MiscEthereum.sol';
+import {AaveSafetyModule} from 'aave-address-book/AaveSafetyModule.sol';
+import {IProposalGenericExecutor} from 'aave-helpers/interfaces/IProposalGenericExecutor.sol';
+
+import {IStakeToken} from './IStakeToken.sol';
+
+/**
+ * @title Amend Safety Module Emissions
+ * @author karpatkey_TokenLogic
+ * - Snapshot: https://snapshot.org/#/aave.eth/proposal/0x0f73500d0f65c72482d352080ea9aa0aa92264eb059b8f646cf6f0e86556bc3d
+ * - Discussion: https://governance.aave.com/t/arfc-amend-safety-module-emissions/16640
+ */
+contract AaveV3Ethereum_AmendSafetyModuleEmissions_20240229 is IProposalGenericExecutor {
+  uint128 public constant AAVE_EMISSION_PER_SECOND_STK_GHO = uint128(100e18) / 1 days; // 100 AAVE per day
+  uint128 public constant AAVE_EMISSION_PER_SECOND_STK_AAVE = uint128(360e18) / 1 days; // 360 AAVE per day
+  uint128 public constant AAVE_EMISSION_PER_SECOND_STK_ABPT = uint128(360e18) / 1 days; // 360 AAVE per day
+
+  function execute() external {
+    // stkGHO
+    IStakeToken.AssetConfigInput[] memory ghoConfigs = new IStakeToken.AssetConfigInput[](1);
+    ghoConfigs[0] = IStakeToken.AssetConfigInput({
+      emissionPerSecond: AAVE_EMISSION_PER_SECOND_STK_GHO,
+      totalStaked: 0, // it's overwritten internally
+      underlyingAsset: AaveSafetyModule.STK_GHO
+    });
+    IStakeToken(AaveSafetyModule.STK_GHO).configureAssets(ghoConfigs);
+
+    // stkAAVE
+    IStakeToken.AssetConfigInput[] memory aaveConfigs = new IStakeToken.AssetConfigInput[](1);
+    aaveConfigs[0] = IStakeToken.AssetConfigInput({
+      emissionPerSecond: AAVE_EMISSION_PER_SECOND_STK_AAVE,
+      totalStaked: 0, // it's overwritten internally
+      underlyingAsset: AaveSafetyModule.STK_AAVE
+    });
+    IStakeToken(AaveSafetyModule.STK_AAVE).configureAssets(aaveConfigs);
+
+    // stkABPT
+    IStakeToken.AssetConfigInput[] memory abptConfigs = new IStakeToken.AssetConfigInput[](1);
+    abptConfigs[0] = IStakeToken.AssetConfigInput({
+      emissionPerSecond: AAVE_EMISSION_PER_SECOND_STK_ABPT,
+      totalStaked: 0, // it's overwritten internally
+      underlyingAsset: AaveSafetyModule.STK_AAVE_WSTETH_BPTV2
+    });
+    IStakeToken(AaveSafetyModule.STK_AAVE_WSTETH_BPTV2).configureAssets(abptConfigs);
+
+    // Allowance for stkGHO to pull funds
+    MiscEthereum.AAVE_ECOSYSTEM_RESERVE_CONTROLLER.approve(
+      MiscEthereum.ECOSYSTEM_RESERVE,
+      AaveV3EthereumAssets.AAVE_UNDERLYING,
+      AaveSafetyModule.STK_GHO,
+      0
+    );
+
+    uint256 distributionDuration = IStakeToken(AaveSafetyModule.STK_GHO).distributionEnd() -
+      block.timestamp;
+    MiscEthereum.AAVE_ECOSYSTEM_RESERVE_CONTROLLER.approve(
+      MiscEthereum.ECOSYSTEM_RESERVE,
+      AaveV3EthereumAssets.AAVE_UNDERLYING,
+      AaveSafetyModule.STK_GHO,
+      AAVE_EMISSION_PER_SECOND_STK_GHO * distributionDuration
```

Comment:
hmm, i think this math is wrong.

What about the pending claims until now?
Assuming the approval was correct before, the correct way would essentially be:
`currentAllowance - distributionDuration*previousEmission + distributionDuration*newEmission`or
`currentAllowance + distributionDuration *(newEmission - previousEmission)` which is the same.

Reply:
alternatively could also be fine to do the math offchain as due to proposal delay would be slightly overestimated.